lib/ansible/module_utils/facts/system/date_time.py

```diff
@@ -49,6 +49,9 @@ def collect(self, module=None, collected_facts=None):
         date_time_facts['epoch'] = now.strftime('%s')
         if date_time_facts['epoch'] == '' or date_time_facts['epoch'][0] == '%':
             date_time_facts['epoch'] = str(int(epoch_ts))
+        date_time_facts['epoch_int'] = str(int(now.strftime('%s')))
```

Comment:
Can you add a comment explaining why this is different from the existing epoch value?

Reply:
This PR and https://github.com/ansible-collections/ansible.windows/pull/123 were created because of this issue: https://github.com/ansible/ansible/issues/72479

Current `epoch` value returns float and string in some circumstances, `epoch_int` is guaranteed to be integer.

Reply:
Sorry, I should have been more specific. Can you add a comment to the code explaining the possibility of the current epoch being a float? That will help anyone in the future looking at the code, since it's otherwise not obvious why we need both epoch and epoch_int.

Reply:
Sure.

Reply:
Fixed in https://github.com/ansible/ansible/pull/73822/commits/d2c161dfff7494e5703158f461ad59a1224b995e.

Reply:
Thanks. This PR is assigned to the 2.12 milestone, which we'll be going through after creating the stable-2.11 branch (which will be done when 2.11 RC1 comes out).

Reply:
Thank you for reviewing!